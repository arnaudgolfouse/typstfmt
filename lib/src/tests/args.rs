use super::make_test;
use crate::Config;

make_test!(call_func_empty, "#f()");
make_test!(call_func_simple, "#f(1,2,3)");
make_test!(parenthesized_not_array, "#(auto)");
make_test!(array_not_parenthesized, "#(auto,)");
make_test!(parenthesized_not_array_break, "#(\nauto)");
make_test!(array_not_parenthesized_break, "#(\nauto,)");
make_test!(
    call_func_long,
    "#f(1,this_is_absurdly_loooooooooong,3)",
    Config {
        max_line_length: 1,
        ..Default::default()
    }
);
make_test!(
    call_func_long_trailing,
    "#f(1,this_is_absurdly_loooooooooong,3,)",
    Config {
        max_line_length: 1,
        ..Default::default()
    }
);
make_test!(
    dont_break_for_one_arg,
    "#f(this_is_absurdly_loooooooooong)",
    Config {
        max_line_length: 1,
        ..Default::default()
    }
);
make_test!(
    dont_break_for_one_arg_with_trail,
    "#f(this_is_absurdly_loooooooooong , )",
    Config {
        max_line_length: 1,
        ..Default::default()
    }
);

make_test!(args_comment_end, ARGS_COMMENT_END);

make_test!(parenth_comment_end, "#(\ntrue// comment\n)");
make_test!(func_comment_end, "#f(\ntrue// comment\n)");

const ARGS_COMMENT_END: &str = "#func(
    ..v_or_hline,
    start: start,
    end: end,
    parent: v_or_hline  // the one that generated this
)";
