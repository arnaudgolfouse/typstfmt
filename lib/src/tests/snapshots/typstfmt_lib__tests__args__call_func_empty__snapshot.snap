---
source: lib/src/tests/args.rs
description: "INPUT\n===\n\"#f()\"\n===\n#f()\n===\nFORMATTED\n===\n#f()"
expression: formatted
---
"#f()"
